### PR TITLE
Add rationale for avoiding storyboards

### PR DESCRIPTION
**Pull request by anandrajeswaran**
Rationale is pretty basic at the moment, but also add links to empty sections that can help justify this and should be written

### CODE REVIEW

File: StoryboardsAndXibs.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Storyboards
+
+## Convention
+In most cases on both macOS and iOS, it is recommended to avoid storyboards (and XIBs) in favor of programmatically creating views and view controllers
```
Comment (markavitale) [inline StoryboardsAndXibs.md:4]: List some notable exceptions like menu bars on macOS and launch screens on iOS?

File: StoryboardsAndXibs.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+## Rationale
+- Data has shown us that disk operations degrade poorly on spinning disks and slower machines. Since we can avoid reading the storyboard/nib from disk at runtiime, we should do so.
```
Comment (markavitale) [inline StoryboardsAndXibs.md:7]: sp: runtiime -> runtime

Comment (markavitale) [inline StoryboardsAndXibs.md:7]: "Data" is really generic, can we specify more directly? Data from profiling performance issues or something? Also "spinning disks and slower machines" phrasing feels very Mac specific, many might read this and say "I'm writing for iOS why should I care?"

File: StoryboardsAndXibs.md (hunk 1 above)
Comment (anandrajeswaran) [reply StoryboardsAndXibs.md:4]: Added a section for Exceptions below

Comment (anandrajeswaran) [reply StoryboardsAndXibs.md:4]: Added an Exceptions section

File: StoryboardsAndXibs.md (hunk 2 above)
Comment (anandrajeswaran) [reply StoryboardsAndXibs.md]: Excellent point. Made a bit more generic since this indeed should be a win for iOS

**Pull request merged**